step-capstone/src/components/TripSettingFormPopover.js

```diff
@@ -0,0 +1,171 @@
+import React, {useState, useEffect} from 'react'
+import {
+  Card,
+  CardActions,
+  CardContent,
+  Button,
+  TextField,
+  Grid,
+} from '@material-ui/core';
+import DateFnsUtils from '@date-io/date-fns';
+import {
+MuiPickersUtilsProvider,
+KeyboardDatePicker,
+} from '@material-ui/pickers';
+import { isValid } from 'date-fns';
+export default class TripSettingFormPopover extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isValidated : false,
+            tripSetting : this.props.tripSetting
+        };
+        this.handleValidation = this.handleValidation.bind(this);
+        this.handleDataChange = this.handleDataChange.bind(this);
+        this.handleSave = this.handleSave.bind(this);
+    }
+    handleValidation(isValidated) {
+        if (this.state.isValidated !== isValidated) {
+            this.setState({isValidated : isValidated});
+        }
+    }
+    handleDataChange(newTripSetting) {
+        this.setState({tripSetting: newTripSetting});
+    }
+    handleSave() {
+        // Doesn't save data if improper input --> doesn't close popover which shows error
```

Comment:
Curious: How does the error get shown?

Reply:
![image](https://user-images.githubusercontent.com/42331170/86639246-9eec8b00-bfa6-11ea-8cdd-3056a23b3c81.png)
It looks like this

Reply:
There seems to be a typo in the screenshot ("Departire" instead of "Departure").

Reply:
Thank you for noticing that!